### PR TITLE
11.0 - Support custom controllers

### PR DESCRIPTION
**Pull request by Pierstoval**
Current TODO (may be subject to change):

- [x] Convert `api.php` to a real Controller.
- [x] Convert `apirest.php` to a real Controller.
- [x] Convert `caldav.php` to a real Controller.
- [ ] Make sure all the above conversions don't use `exit` or `die` so that a proper `Response` object can be used.
- [x] Make a "SecurityStrategy" annotation.
- [x] Make it load a custom Request attribute during the Routing compilation process.
- [x] Make this request attribute change the actual security strategy via a Kernel Request listener.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -109,7 +109,8 @@
         "squizlabs/php_codesniffer": "^3.10",
         "symfony/browser-kit": "^6.4",
         "symfony/http-client": "^6.4",
-        "symfony/var-dumper": "^6.4"
+        "symfony/var-dumper": "^6.4",
+        "symfony/web-profiler-bundle": "^7.1"
```
Comment (cedric-anne) [inline composer.json:113]:
```suggestion
        "symfony/web-profiler-bundle": "^6.4"
```

File: dependency_injection/services.php [hunk 2]
```diff
@@ -40,6 +40,12 @@
 return static function (ContainerConfigurator $container): void {
     $projectDir = dirname(__DIR__);
 
+    $parameters = $container->parameters();
+
+    // Default secret, just in case
+    $parameters->set('env(APP_SECRET)', uniqid('', true));
+    $parameters->set('kernel.secret', env('APP_SECRET'));
```
Comment (cedric-anne) [inline dependency_injection/services.php:47]: There is a `glpicrypt.key` file in the config dir that is generated during installation/update. We should rely on it, I guess.
See `GLPIKey` class.

**Review by cedric-anne — COMMENTED** [review]
Adding the profiler bundle could be done in a separate PR. I guess it could be merged quickly.

**Comment by cconard96** [comment]
The two APIs seem broken. Getting 404 error HTML pages as a response for all checked API endpoints.

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**